src/table/utils/color-utils.js

```diff
@@ -34,25 +43,33 @@ export function resolveExpression(input) {
 }
 
 export function isDarkColor(color) {
-  let r;
-  let g;
-  let b;
-  let matches;
-  if (
-    (matches =
-      /^rgb\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*\)$/i.exec(color) ||
-      /^rgba\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d(\.\d+)?)\s*\)$/i.exec(color))
-  ) {
-    r = parseInt(matches[1], 10);
-    g = parseInt(matches[2], 10);
-    b = parseInt(matches[3], 10);
-  } else if ((matches = /^#([A-f0-9]{2})([A-f0-9]{2})([A-f0-9]{2})$/i.exec(color))) {
-    // #aBc123
-    r = parseInt(matches[1], 16);
-    g = parseInt(matches[2], 16);
-    b = parseInt(matches[3], 16);
-  }
+  const rgba = resolveExpression(color);
+  const matches =
+    /^rgb\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*\)$/i.exec(rgba) ||
+    /^rgba\(\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d{1,3})\s*,\s*(\d(\.\d+)?)\s*\)$/i.exec(rgba);
+  const r = +matches?.[1];
+  const g = +matches?.[2];
+  const b = +matches?.[3];
+
   // Using the HSP (Highly Sensitive Poo) value, determine whether the color is light or dark
+  // the equation is from http://alienryderflex.com/hsp.html
+  const HSP = Math.sqrt(0.299 * (r * r) + 0.587 * (g * g) + 0.114 * (b * b));
```

Comment:
Let's not change this. This is shared between charts, so even if this is more accurate, we should review this when we move this to nebula or similar

Reply:
Sure